crates/networking/p2p/src/network.rs

```diff
@@ -459,10 +462,18 @@ impl Network {
             }
             SwarmEvent::Behaviour(behaviour_event) => match behaviour_event {
                 ReamBehaviourEvent::Identify(_) => None,
-                ReamBehaviourEvent::Discovery(DiscoveredPeers { peers }) => {
+                ReamBehaviourEvent::Discovery(DiscoveryOutEvent::DiscoveredPeers(
+                    DiscoveredPeers { peers },
+                )) => {
                     self.handle_discovered_peers(peers);
                     None
                 }
+                ReamBehaviourEvent::Discovery(DiscoveryOutEvent::UpdatedEnr(UpdatedEnr {
+                    enr,
+                })) => {
+                    *self.network_state.local_enr.write() = enr;
+                    None
+                }
```

Comment:
Can you just do a match case instead of duplicating ``ReamBehaviourEvent::Discovery`` twice